tests/System.Slices.Tests/MemoryTests.cs

```diff
@@ -30,25 +31,72 @@ public void SimpleTestS()
         [Fact]
         public void ArrayMemoryLifetime()
         {
-            Memory<byte> copyStoredForLater;
-            using (var owner = new OwnedArray<byte>(1024)) {
-                Memory<byte> memory = owner.Memory;
-                Memory<byte> memorySlice = memory.Slice(10);
-                copyStoredForLater = memorySlice;
-                using (memorySlice.Reserve()) { // increments the "outstanding span" refcount
-                    Assert.Throws<InvalidOperationException>(() => { // memory is reserved; cannot dispose
-                        owner.Dispose();
-                    });
-                    Span<byte> span = memorySlice.Span;
-                    span[0] = 255;
-                } // releases the refcount
-            }
+            Memory<byte> copyStoredForLater = default(Memory<byte>);
+            Assert.Throws<InvalidOperationException>(() => {
```

Comment:
It is the implicit
```csharp
finally {
    owner.Dispose();
}
```
That is throwing the exception.  I could unpack the `using` statement?

If I disable the check for double calls to Dispose, then this would be reverted.  And other code does do double dispose on subclasses, which is why the tests are currently failing.